mcstatus/protocol/connection.py

```diff
@@ -448,6 +448,9 @@ def __init__(self) -> None:
 
     def read(self, length: int) -> bytearray:
         """Return :attr:`.received` up to length bytes, then cut received up to that point."""
+        if len(self.received) == 0:
+            raise IndexError("No data to read!")
```

Comment:
While `IndexError` is indeed what would be raised if this happened without handling, do we really want to use this here? Wouldn't something like say an `IOError` make more sense? Indicating that there was an error in getting the output from the server.

Reply:
This is what raising other classes, I though Liskov substitution principle should be applied here too.

Reply:
Hm? What other classes? For example in `TCPSocketConnection`, the `read` method is already using `IOError`, somewhere, same thing for read_varint when the data isn't complete.

Reply:
Changed here [here](https://github.com/py-mine/mcstatus/compare/77dfd352e41f95670578b9333ca511926d599420..9a6f4ad653e5ccc77ad0937407aa5f6f68bf35a1).